src/util/PopulationsMap.ts

```diff
@@ -439,19 +442,128 @@ export const mapExistingTestCase = (existingTestCasePC, group: Group) => {
     group?.measureObservations?.length > 0
   ) {
     if (isBooleanOrPatientBasis) {
-      return addDefaultObservationsForBooleanAndPatientBasedTestcases(
+      return addDefaultObservationsForPatientBasedTestcases(
         existingTestCasePC,
         group
       );
     } else {
-      // need to implement for non-boolean and episode based measures
-      return { ...existingTestCasePC };
+      return addDefaultObservationsForEpisodeBasedTestcases(
+        existingTestCasePC,
+        group
+      );
+    }
+  }
+  return { ...existingTestCasePC };
+};
+
+const countObservations = (
+  expectedRelatedPopulationValue,
+  expectedRelatedPopulationExclusionValue
+) => {
+  if (
+    expectedRelatedPopulationValue &&
+    expectedRelatedPopulationExclusionValue
+  ) {
+    const observationsCount = Math.max(
+      expectedRelatedPopulationValue - expectedRelatedPopulationExclusionValue,
+      0
+    );
+    return observationsCount;
+  }
+  return expectedRelatedPopulationValue;
+};
+
+const addDefaultObservationsForEpisodeBasedTestcases = (
+  existingTestCasePC,
+  group
+) => {
+  const findPopulation = (name) =>
+    existingTestCasePC.populationValues.find(
+      (popVal) => popVal.name === name && popVal.expected > 0
```

Comment:
Is 0 considered to be the default ? I believe that, User can expect a population to return 0 elements.

Reply:
In this case, I am only concerned with populations where denominator, denominatorExclusion, numerator, numertorExclusions expected values are greater than 0, this is because only in those conditions only we would be adding the default observations or else you don't need them.